src/combinatorics/simple_graph/basic.lean

```diff
@@ -476,6 +476,87 @@ begin
   tauto,
 end
 
+/-lemma finset.compl_to_finset (α : Type u) [decidable_eq α] [fintype α] (s : set α) :
+  sᶜ.to_finset = (s.to_finset)ᶜ :=
+by ext; simp-/
+
+variables [decidable_eq V]
+
+lemma neighbor_set_union_compl_neighbor_set_card [fintype V] (G : simple_graph V)
+  [decidable_rel G.adj] [decidable_rel Gᶜ.adj] (v : V) :
+  fintype.card ((G.neighbor_set v ∪ Gᶜ.neighbor_set v) : set V) = fintype.card V - 1 :=
+begin
+  -- i don't love the `fintype.subtype_of_fintype situation` i have here
+  classical,
+  simp_rw neighbor_set_union_compl_neighbor_set_eq,
+  rw ← set.to_finset_card,
+  simp,
+  rw fintype.card_of_finset' {v}ᶜ,
+  rw card_compl,
+  rw ← card_singleton v,
+  simp,
+end
+
+lemma compl_neighbor_set (G : simple_graph V) [decidable_rel G.adj] (v : V) :
+  Gᶜ.neighbor_set v = (G.neighbor_set v)ᶜ \ {v} :=
+begin
+  ext w,
+  simp,
+  exact ⟨λ ⟨hne, hnadj⟩, ⟨hnadj, ne.symm hne⟩, λ ⟨hnadj, hne⟩, ⟨ne.symm hne, hnadj⟩⟩,
+end
+
+lemma compl_neighbor_set' (G : simple_graph V) [decidable_rel G.adj] (v : V) :
+  Gᶜ.neighbor_set v = set.univ \ (G.neighbor_set v ∪ {v}) :=
+begin
+  ext w,
+  simp,
+  push_neg,
+  exact ⟨λ ⟨hne, hnadj⟩, ⟨ne.symm hne, hnadj⟩, λ ⟨hne, hnadj⟩, ⟨ne.symm hne, hnadj⟩⟩,
+end
+
+lemma card_compl_neighbor_set' [fintype V] (G : simple_graph V) [decidable_rel G.adj] (v : V) :
+  fintype.card (Gᶜ.neighbor_set v) = fintype.card V - fintype.card (G.neighbor_set v) - 1 :=
+begin
+  -- rw compl_neighbor_set' G v,
```

Comment:
I get a "motive is not type correct" error when I do `rw compl_neighbor_set G v` and `rw compl_neighbor_set' G v`. Am I doing something wrong here?

Reply:
You can `simp_rw` here.  The problem is that `fintype.card` has a secret argument which is the `fintype` instance, and `rw` isn't able to rewrite multiple things simultaneously.

Another way around this is to use `finset`s directly, for example `neighbor_finset`.  They're easier to rewrite than `fintype.card` of a set.